### PR TITLE
Zookeeper reference implementation cleanup

### PR DESCRIPTION
**Pull request by stanlemon**
Most of this is simple cleanup. There is one notable change though, in the original reference implementation for 0.10 the zookeeper watch supports multiple roots. I don't recall what the reason for this complexity was, probably accounting for the weird way we tried handling global sidelines in the implementation that preceded this. I am confident that there's a better/cleaner way of doing that. I am just not sure that additional complexity is worth it.

### CODE REVIEW

File: src/main/java/com/salesforce/storm/spout/sideline/recipes/trigger/zookeeper/ZookeeperWatchTrigger.java [hunk 1]
```diff
@@ -112,19 +115,11 @@
         .setDateFormat("yyyy-MM-dd HH:mm:ss")
         .create();
 
-    /**
```
Comment (stanlemon) [inline src/main/java/com/salesforce/storm/spout/sideline/recipes/trigger/zookeeper/ZookeeperWatchTrigger.java]: I know this seems weird, but I prefer the docs from the interface in situations like this where they're just copy pasted.

File: src/main/java/com/salesforce/storm/spout/dynamic/JSON.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import com.google.gson.GsonBuilder;
+import com.salesforce.storm.spout.sideline.persistence.FilterChainStepTypeAdapterFactory;
+
+/**
+ * Thin wrapper around JSON parsing.
+ *
+ * Intended to hide the particular JSON implementation of the day.
+ */
+@SuppressWarnings("checkstyle:AbbreviationAsWordInName")
+public class JSON {
```
Comment (Crim) [inline src/main/java/com/salesforce/storm/spout/dynamic/JSON.java]: Json instead of all caps?  Does this pass checkstyle?

Comment (stanlemon) [reply src/main/java/com/salesforce/storm/spout/dynamic/JSON.java]: See the annotation?

Comment (Crim) [reply src/main/java/com/salesforce/storm/spout/dynamic/JSON.java]: Seems like you violated checkstyle

Comment (stanlemon) [reply src/main/java/com/salesforce/storm/spout/dynamic/JSON.java]: Yes. But with malice and forethought.

**Pull request merged**